pkg/scheduler/service/transition.go

```diff
@@ -80,26 +90,22 @@ func (t *ClusterStatusTransition) StartReconciliation(runtimeID string, configVe
 			newClusterState.Cluster.RuntimeID, model.ClusterStatusReconciling)
 
 		//create reconciliation entity
-		reconEntity, err := t.reconRepo.CreateReconciliation(newClusterState, preComponents)
+		reconEntity, err := reconRepo.CreateReconciliation(newClusterState, preComponents)
 		if err == nil {
 			t.logger.Infof("Starting reconciliation for cluster '%s' succeeded: reconciliation successfully enqueued "+
 				"(scheudlingID: %s)", newClusterState.Cluster.RuntimeID, reconEntity.SchedulingID)
 			return nil
 		}
 
-		//check for cluster configuration issue
 		if reconciliation.IsEmptyComponentsReconciliationError(err) {
 			t.logger.Errorf("Cluster transition tried to add cluster '%s' to reconciliation queue but "+
 				"cluster has no components", newClusterState.Cluster.RuntimeID)
-
-			//set cluster status to non-recoverable error
-			//TODO: throw a dedicated error and update status ouside (db chnage will be rolled back after #507 is fixed)
-			if _, updateErr := t.inventory.UpdateStatus(newClusterState, model.ClusterStatusReconcileError); updateErr != nil {
+			_, updateErr := inventory.UpdateStatus(newClusterState, model.ClusterStatusReconcileError)
```

Comment:
That won'T work.. if an error occures and we are returning the error, also this call will be reverted.

This operation has to happen outside of the TX dependant if the TX was returning an error which matches `reconciliation.IsEmptyComponentsReconciliationError(err) `

Reply:
Makes sense! Done